contracts/src/v0.8/l2ep/test/v1_0_0/arbitrum/ArbitrumCrossDomainForwarder.t.sol

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {ArbitrumCrossDomainForwarder} from "../../../dev/arbitrum/ArbitrumCrossDomainForwarder.sol";
+import {Greeter} from "../../../../../v0.8/tests/Greeter.sol";
+import {L2EPTest} from "../L2EPTest.t.sol";
+
+// Use this command from the /contracts directory to run this test file:
+//
+//  FOUNDRY_PROFILE=l2ep forge test -vvv --match-path ./src/v0.8/l2ep/test/v1_0_0/arbitrum/ArbitrumCrossDomainForwarder.t.sol
+//
+contract ArbitrumCrossDomainForwarderTest is L2EPTest {
+  /// Helper variables
+  address internal s_strangerAddr = vm.addr(0x1);
+  address internal s_l1OwnerAddr = vm.addr(0x2);
+  address internal s_crossDomainMessengerAddr = toArbitrumL2AliasAddress(s_l1OwnerAddr);
+  address internal s_newOwnerCrossDomainMessengerAddr = toArbitrumL2AliasAddress(s_strangerAddr);
+
+  /// Contracts
+  ArbitrumCrossDomainForwarder internal s_arbitrumCrossDomainForwarder;
+  Greeter internal s_greeter;
+
+  /// Events
+  event L1OwnershipTransferRequested(address indexed from, address indexed to);
+  event L1OwnershipTransferred(address indexed from, address indexed to);
+
+  /// Setup
+  function setUp() public {
+    // Deploys contracts
+    vm.startPrank(s_l1OwnerAddr, s_l1OwnerAddr);
+    s_arbitrumCrossDomainForwarder = new ArbitrumCrossDomainForwarder(s_l1OwnerAddr);
+    s_greeter = new Greeter(address(s_arbitrumCrossDomainForwarder));
+    vm.stopPrank();
+  }
+}
+
+contract Constructor is ArbitrumCrossDomainForwarderTest {
```

Comment:
I would name this <contract>_<function> so the gas snapshot and CLI are more readable

Reply:
Ah I see what you mean 

It looks like `forge snapshot` doesn't include the path to the file or any information about the parent contract name, so you just get a bunch of duplicate lines

If I change the contract name to _, I run into an error, so for now I've prefixed the tests with a more specific name - lmk what you think!

Reply:
Oh that was a formatting issue. I meant to type <contract>_<function> as name